org.envirocar.app/src/org/envirocar/app/views/obdselection/OBDSelectionActivity.java

```diff
@@ -103,12 +103,12 @@ private void setupBluetoothSwitch() {
         mEnableBTText.setText(isBTEnabled ?
                 R.string.obd_selection_bluetooth_on :
                 R.string.obd_selection_bluetooth_off);
-        mSwitch.setChecked(mBluetoothHandler.isBluetoothEnabled());
 
         mSwitch.setOnCheckedChangeListener((buttonView, isChecked) -> {
             if (isChecked) {
-                //                mSwitch.setChecked(false);
+                //  If Bluetooth is allowed then mSwitch checked
                 mBluetoothHandler.enableBluetooth(OBDSelectionActivity.this);
+                mSwitch.setChecked(mBluetoothHandler.isBluetoothEnabled());
             } else {
                 mBluetoothHandler.disableBluetooth(OBDSelectionActivity.this);
```

Comment:
You should also consider the other way: if you decline deactivating Bluetooth, the toggle button also changes its stage, which should not happen.

Reply:
I see no request when we turn off the Bluetooth. Updated accordingly `mSwitch.setChecked(false)`